plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4251,6 +4257,34 @@ public void rebootVM(final Connection conn, final VM vm, final String vmName) th
         }
     }
 
+    protected String skipOrRemoveSR(Connection conn, SR sr) {
+        if (sr == null) {
+            return null;
+        }
+        if (s_logger.isDebugEnabled()) {
+            s_logger.debug(logX(sr, "Removing SR"));
+        }
+        try {
+            Set<VDI> vdis = sr.getVDIs(conn);
+            for (VDI vdi : vdis) {
+                Map<java.lang.String, Types.VdiOperations> currentOperation = vdi.getCurrentOperations(conn);
+                if (currentOperation == null || currentOperation.size() == 0) {
+                    continue;
+                }
+                return null;
+            }
+            removeSR(conn, sr);
+            return null;
+        } catch (XenAPIException e) {
+            s_logger.warn(logX(sr, "Unable to get current opertions " + e.toString()), e);
+        } catch (XmlRpcException e) {
+            s_logger.warn(logX(sr, "Unable to get current opertions " + e.getMessage()), e);
+        }
+        String msg = "Remove SR failed";
+        s_logger.warn(msg);
+        return msg;
```

Comment:
@DaanHoogland, sorry to not share your opinion, but, since ACS is open-source I think that "JavaDocing" could be very handy and turns things more organized when needed.
Those methods like "geters" and "seters" may not received a javadoc because they are self-explanatory. But since others people will read your code in the future, it may worth wasting some time JavaDocing the method. That will take you just a few minutes to do it and it is better than letting other people wasting much more time to find out what that code do or how it works.